src/core/qgsvectorlayercache.cpp

```diff
@@ -58,7 +58,9 @@ int QgsVectorLayerCache::cacheSize()
 
 void QgsVectorLayerCache::setCacheGeometry( bool cacheGeometry )
 {
-  mCacheGeometry = cacheGeometry && mLayer->hasGeometryType();
+  bool shouldCache = cacheGeometry && mLayer->hasGeometryType();
```

Comment:
Can you rename the local var to `shouldCacheGeom`, will save me a bit of thinking next time ;)